src/migration-scripts/openconnect/2-to-3

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env python3
+#
+# Copyright (C) 2022 VyOS maintainers and contributors
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 2 or later as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+# T4982: Retain prior default TLS version (v1.0) when upgrading installations with existing openconnect configurations
+
+import sys
+
+from vyos.configtree import ConfigTree
+
+if len(sys.argv) < 2:
+    print("Must specify file name!")
+    sys.exit(1)
+
+file_name = sys.argv[1]
+
+with open(file_name, 'r') as f:
+    config_file = f.read()
+
+config = ConfigTree(config_file)
+cfg_base = ['vpn', 'openconnect']
+
+if config.exists(cfg_base):
```

Comment:
You're logic should rather be:

```python3
config = ConfigTree(config_file)
cfg_base = ['vpn', 'openconnect']

# bail out early if service is unconfigured
if not config.exists(cfg_base):
    sys.exit(0)

# new default is TLS 1.2 - set explicit old default value of TLS 1.0 for upgraded configurations to keep compatibility
tls_min_path = cfg_base + ['tls-version-min']
if not config.exists(tls_min_path):
    config.set(tls_min_path, value='1.0')

try:
    with open(file_name, 'w') as f:
        f.write(config.to_string())
except OSError as e:
    print("Failed to save the modified config: {}".format(e))
    sys.exit(1)
```

Reply:
```python
if not config.exists(tls_min_path):
```
No idea how I managed to forget this check! 😲